### PR TITLE
feat: Refactor existing `parseBoxShadow` to parse both box and text shadows

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
## Description

`box-shadow` and `text-shadow` both share the same syntax signature.
https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow#syntax
https://developer.mozilla.org/en-US/docs/Web/CSS/text-shadow#syntax

The only difference is `text-shadow` don't have `inset` and `spread` values. And added a extractor which can be used in future for text-shadow UI.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [x] added tests

### CODE REVIEW

File: packages/css-data/src/property-parsers/__snapshots__/shadows.test.ts.snap
Comment (kof) [inline packages/css-data/src/property-parsers/__snapshots__/shadows.test.ts.snap:1]: @TrySound what's your opinion on shapshots, I wonder if we had switched to inline snapshots

File: packages/css-data/src/property-parsers/box-shadow-property-extractor.test.ts
Comment (kof) [inline packages/css-data/src/property-parsers/box-shadow-property-extractor.test.ts:3]: I suggest to import specifically the file instead of the index

File: packages/css-data/src/property-parsers/text-shadow-property-extractor.ts
Comment (kof) [inline packages/css-data/src/property-parsers/text-shadow-property-extractor.ts:22]: is there any way to not use null? we are generally trying to use undefined all the time and null only where some external api forces us to use null

**Review by kof — APPROVED** [review]
I am approving with a few minor comments, but lets get an approval from @TrySound

File: packages/css-data/src/property-parsers/text-shadow-property-extractor.ts
Comment (JayaKrishnaNamburu) [reply packages/css-data/src/property-parsers/text-shadow-property-extractor.ts:22]: Will switch to `undefined` then 👍

File: packages/css-data/src/property-parsers/__snapshots__/shadows.test.ts.snap
Comment (TrySound) [reply packages/css-data/src/property-parsers/__snapshots__/shadows.test.ts.snap:1]: yep, we use only inline snapshots

File: packages/css-data/src/property-parsers/text-shadow-property-extractor.ts
Comment (TrySound) [inline packages/css-data/src/property-parsers/text-shadow-property-extractor.ts:28]: is it different from box shadow extractor?

Comment (JayaKrishnaNamburu) [reply packages/css-data/src/property-parsers/text-shadow-property-extractor.ts:28]: Yes, `text-shadow` don't have `blur` and `spread`. And really didn't want to mix the extractors just in case if we need to modify them if needed too.

Comment (JayaKrishnaNamburu) [reply packages/css-data/src/property-parsers/text-shadow-property-extractor.ts]: @TrySound merged both the extractors now. Let's see if we really need to be separated down the line then.

**Review by TrySound — APPROVED** [review]

**Pull request merged**
